app.js

```diff
@@ -219,28 +222,58 @@ app.post('/interactions', verifyKeyMiddleware(process.env.PUBLIC_KEY), async (re
             };
 
             const lbargs = args[0].split('+');
-            if (lbargs[1] ==='totd') {
-                track_info.endTimestamp = Number(convertNumberToBase(lbargs[2], 64, 10));
-            }
-            if (lbargs[lbargs.length - 1] === 'f') {
-                args.push(0);
-            } else if (lbargs[lbargs.length - 1] === 'l') {
-                args.push(1000-args[3]);
-            } else if (lbargs[lbargs.length - 1] === 'p') {
-                data.values[0].split(';').forEach((e) => {
-                    args.push(e);
-                });
-            }
+            if (lbargs[1] === 'totd') track_info.endTimestamp = Number(convertNumberToBase(lbargs[2], 64, 10));
+            let body;
+            if (lbargs[lbargs.length - 1] ==='i') {
```

Comment:
Can we add some deserialization logic to leaderboard arguments to make this code a little easier to follow? I see you're fetching arguments relative to the end of the array length, presumably because it's become a PITA (hey louis) to use this interface.

e.g. [these arguments](https://github.com/Khujou/trackmania-bot/blob/237642df7462c893af3320712e6527199cbe923f/trackmania.js#L771) could become something like this:
```js
{
  timestamp: decode(lbargs[0]),
  mapType: decode(lbargs[1]),
  mapId: decode(lbargs[2]),
  command: decode(lbargs[3),
}
```

This way, you can write some `encodeargs` and `decodeargs` which actually care about the indices / encoding, and the rest of your code can just use the field names instead:
```js
lbargs.command === LB_COMMAND_FIRST // lbargs[lbargs.length-1] === 'i'
lbargs.mapId // lbargs[lbargs.length-2]
```

This would make code reviews (and just reading + maintaining the code in general) a little easier. I had to hunt for the customId code and read that to understand what's even in `lbargs`. If it weren't an array, we wouldn't need to memorize bits of the codebase just to understand what this small chunk is doing

Reply:
new commit i think i did what you asked